### PR TITLE
Scala Native and Scala 3

### PR DESCRIPTION
**Pull request by sideeffffect**
closes https://github.com/cquiroz/scala-java-locales/pull/340
closes https://github.com/cquiroz/scala-java-locales/pull/338

~blocked on https://github.com/cquiroz/sbt-locales/pull/190~
Blocked on a new version release with this changeset https://github.com/scalameta/munit/pull/477

### CODE REVIEW

**Comment by lolgab** [comment]
sbt-locales blocked seems to be merged and released.
Can this move forward @sideeffffect ?

**Comment by cquiroz** [comment]
I think this is still waiting for a munit release?

**Comment by sideeffffect** [comment]
@lolgab I have no idea where I should put more explicit type annotations... `generatePattern` is a big function :disappointed:

**Comment by lolgab** [comment]
@sideeffffect I would start by annotating all the functions to see if it makes a difference.

**Comment by lolgab** [comment]
I guess you can remove the `[WIP]` now :-)

**Comment by sideeffffect** [comment]
Actually no, this PR still depends on munit `1.0.0-M3+39-d7ab5753-SNAPSHOT`
Let's see when they will release a regular version (still probably Milestone, but that should be OK for us here)
https://discord.com/channels/632642981228314653/671278431102697472/975183987158958121

**Comment by sideeffffect** [comment]
This is ready to be merged @cquiroz :tada:

**Comment by cquiroz** [comment]
Do you need sbt-locales?
https://github.com/cquiroz/sbt-locales/releases/tag/v3.2.0

**Comment by sideeffffect** [comment]
We can do without sbt-locales 3.2.0 until you fix the PGP key for it's publishing.
This is good to be merged as it is.

File: core/native/src/main/scala-3/java/text/Normalizer.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    !(cstr + i.toLong) = 0.toByte
+
+    cstr
+  }
+
+}
+
+@link("utf8proc")
+@extern
```
Comment (cquiroz) [inline core/native/src/main/scala-3/java/text/Normalizer.scala:59]: Is this a scala-native specific feature?

Comment (sideeffffect) [reply core/native/src/main/scala-3/java/text/Normalizer.scala:59]: No idea. I just copied and adjusted whatever was in Scala 2 variant of Scala Native part of the codebase :wink:

Comment (sideeffffect) [reply core/native/src/main/scala-3/java/text/Normalizer.scala:59]: Original code was merged as part of https://github.com/cquiroz/scala-java-locales/pull/356

**Review by cquiroz — APPROVED** [review]
LGTM, thanks a lot

**Pull request merged**